container/writer.go

```diff
@@ -96,6 +96,10 @@ func (avroWriter *Writer) WriteRecord(record AvroRecord) error {
 }
 
 func (avroWriter *Writer) Flush() error {
+	if !avroWriter.headerWritten {
+		return nil
+	}
```

Comment:
We do things like close and reset the writer below. Do we not need to do that still?